File: backend/src/main/java/org/yougrow/backend/entities/projection/Question/QuestionDetail.java
package org.yougrow.backend.entities.projection.Question;

import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import org.yougrow.backend.entities.projection.IdAndName;

import java.util.List;

@JsonPropertyOrder({
        "id",
        "question",
        "correctAnswer",
        "questionType",
        "answerA",
        "answerB",
        "answerC",
        "answerD",
        "quizzes",
        "isFavorite",
        "tags"
})
public interface QuestionDetail {
    Long getId();

    String getQuestion();

    String getQuestionType();

    String getCorrectAnswer();

    String getAnswerA();

    String getAnswerB();

    String getAnswerC();

    String getAnswerD();

    Boolean getIsFavorite();

    List<IdAndName> getTags();

    List<IdAndName> getQuizzes();
}

File: frontend/src/app/interfaces/questionDetail.ts
import {QuizShort} from "./quizShort";
import {TagShort} from "./tagShort";

export interface QuestionDetail {
  id?: number;
  question: string;
  password?: string;
  questionType: string;
  correctAnswer: string;
  answerA?: string;
  answerB?: string;
  answerC?: string;
  answerD?: string;
  isFavorite: boolean;
  quizzes: QuizShort[];
  tags: TagShort[];
}
